qiskit_ibm_runtime/options/utils.py

```diff
@@ -92,8 +92,8 @@ def __new__(cls, *_, **kwargs):  # type: ignore
             orig_field_names.add(fld.name)
 
         for key, val in kwargs.items():
-            if key not in orig_field_names:
-                all_fields.append((key, type(val), field(default=None)))
+            if key in dir(cls):
+                setattr(cls, key, val)
```

Comment:
While this accomplishes the desired output, creating a new dataclass (line 98) is unnecessary if we are not allowing arbitrary kwargs. I think it'd be worth understanding what this decorator does to fix the issue properly.